### ISSUE TITLE
Terragrunt to read account IDs & co. from the tfvars file

### ISSUE THREAD

**Issue by P0NDER0SA** [cds-snc/notification-planning-core#410]
## Description

As a Developer,
I need to be able to use Terraform variables in code such as the AWS account ID,
so that I can achieve secure and maintainable code. 

## WHY are we building? 

Code security and maintainability.

## WHAT are we building?

* Variables in code from Terraform can be read from Terragrunt, such as AWS Account IDs.
* Proof of concept with having Terragrunt reading from the tfvars files, consolidating the account ID in one of these. This should be done with minimal effort. The #109 task would tackle the next big steps.
* We need to have Terragrunt reading the tfvars.

## VALUE created by our solution

Secure and consistent/debuggable code.

## QA Steps

- [x] Search across entire TF repository for hardcoded AWS account ids
- [x] All Plans and Applies are working against Staging and Production
- [x] Verify that the account ID is consolidated within one tfvars file, read from TG and demonstrated as a proof of concept for #109.

## Acceptance Criteria

- [ ] **Scenario: Tokenizing TF variables such as AWS Account IDs in Terraform Code**
  - Given the issue is about tokenizing AWS Account IDs in Terraform code
  - Account IDs are stored in a tfvars file.
  - No AWS Account Ids are visible in code repo

**Comment by P0NDER0SA** [comment]
Identified a list of all of the AWS account ids we have in our Terraform code.  Investigated the best way to set up variables for them -- decided that we should focus on the CDS internal ids (as many of the other external ones are public anyways).  

In order to use TFVars variables for these we really need to adjust our code to allow Terragrunt to use our TFVars file (which is scoped to Terraform for some silly reason).  

Discovered a newer function that could help us do just that-- need to compare this with Bens other env enhancements and decide how to proceed.  Also if we use this, we will have to update our actions to use a newer version of the Terragrunt Actions.

**Comment by P0NDER0SA** [comment]
https://github.com/cds-snc/notification-terraform/pull/1498

**Comment by P0NDER0SA** [comment]
I have a working model for this on Dev, but i need to get it workign on Staging before I can merge it.

**Comment by P0NDER0SA** [comment]
Blocked by an issue with GHA that i can't reproduce locally.  Will bug Ben when he's available.  Updated the code to use variables (on dev and staging and sandbox environments).  

Need to fix the plans for Staging before devising a way to keep production working as it does until i can get staging 100%.

https://github.com/cds-snc/notification-terraform/pull/1516

**Comment by P0NDER0SA** [comment]
Ben and I want to debug this more and find out why the planning is failing without any errors/warnings today

**Comment by ben851** [comment]
For dev TF Plan github action, I just switched the role to notification-terraform-apply so that the DNS management works.

**Comment by P0NDER0SA** [comment]
amalgamated all configuraable values in staging into our tfvars file (this was a big job!) and created all variable declarations in our top level Terragrunt hcl file.  Currently working on removing redundant variable declarations in downstream variable files for staging.  We very much underestimated this one!

**Comment by P0NDER0SA** [comment]
all variables have been pulled together for dev and staging and are in 1pass tfvars files.  they are extensive.  Plans are working against dev and staging -- we need to verify the plans to make sure they aren't doing anythign dangerous (the new sensitive values are causing it to show a good many "changes" that might not really be changes).  Making very good progresss.

**Comment by sastels** [comment]
Working through applies and debugging. Some issues arose on dev (VPN randomly breaking).

**Comment by sastels** [comment]
VPN working abain on dev :tada: Next:
- apply new code to dev and verify no strange changes
- look at staging.

**Comment by sastels** [comment]
Gonna look at the prod plan first.

**Comment by jimleroyer** [comment]
Today

* we want to apply the plan in staging.
* run the plan against production.
* find discrepancies between GitHub passwords and 1password passwords.

**Comment by P0NDER0SA** [comment]
new PR has been created with minimal commits!
https://github.com/cds-snc/notification-terraform/pull/1562

**Comment by jimleroyer** [comment]
* Let's create a revert PR for that giant PR in case that something happens and we need a quick rollback. The changes in staging will be sitting for the next few days. 
* We need all the tfvars for the production environment.

**Comment by P0NDER0SA** [comment]
Merged!  It's deployed on staging.

One small issue with Heartbeat - just an out of date config value for the API key.  Fixed!

We will re-create a new api key for production before we release to nip this in the bud.  

TFVars for prod have been reviewed.  One more pass on this today and a Plan against production before release!

**Comment by P0NDER0SA** [comment]
Looking good in Production!

**Comment by sastels** [comment]
account ids in:
- check_rds_cluster_updates.yml (can pull from github secrets)
- terragrunt_destroy_environment.yml (can pull from github secrets)
- file_scanning.tf (Steve will add these!)
- ec2.tf
- sentinel.tf
- ecr_user.tf
- aws_integration.tf (new relic)
-

**Comment by sastels** [comment]
PR to [Move out a couple account ids](https://github.com/cds-snc/notification-terraform/pull/1578)

**Comment by sastels** [comment]
I'll do another check today to double check that we got everything.

**Comment by sastels** [comment]
dang it, [I missed one
](https://github.com/cds-snc/notification-terraform/blob/8f710cc7520da0a789ac450a3062f7fc67ca9a67/.github/workflows/terragrunt_destroy_environment.yml#L61).
PR [here](https://github.com/cds-snc/notification-terraform/pull/1585)

**Comment by sastels** [comment]
All good now!
